tests/testthat/test-qenv-replace_code.R

```diff
@@ -0,0 +1,63 @@
+q <- new_qenv()
+qq <- within(q, i <- iris)
+
+# argument checks ----
+test_that("object argument accepts qenv", {
+  testthat::expect_no_error(replace_code(qq, "i <- mtcars"))
+})
+
+test_that("object argument accepts qenv.error", {
+  qe <- within(q, stop())
+  testthat::expect_no_error(replace_code(qe, "i <- mtcars"))
+})
+
+test_that("code argument accepts character", {
+  testthat::expect_no_error(replace_code(qq, "i <- mtcars"))
+})
+
+test_that("code argument accepts language", {
+  testthat::expect_no_error(replace_code(qq, quote(i <- mtcars)))
+})
+
+test_that("code argument accepts expression", {
+  testthat::expect_no_error(replace_code(qq, expression(i <- mtcars)))
+})
+
+# code replacement ----
+testthat::test_that("length of @code is unchanged", {
+  qr <- replace_code(qq, "i <- mtcars")
+  testthat::expect_identical(length(get_code(qq)), length(get_code(qr)))
+})
+
+testthat::test_that("contents of @code are changed", {
+  qr <- replace_code(qq, "i <- mtcars")
+  testthat::expect_failure(testthat::expect_identical(get_code(qq), get_code(qr)))
+})
+
+testthat::test_that("code passed to replace_code is placed in @code", {
+  replacement <- "i <- mtcars"
+  qr <- replace_code(qq, replacement)
+  testthat::expect_true(replacement %in% get_code(qr))
+})
+
+testthat::test_that("code passed to replace_code is replaces last element of @code", {
+  qq <- within(qq, m <- mtcars)
+  replacement <- "i <- mtcars"
+  qr <- replace_code(qq, replacement)
+  testthat::expect_identical(replacement, get_code(qr)[length(get_code(qr))])
+})
+
+testthat::test_that("non-last elements of @code are unaffected", {
+  qq <- within(qq, m <- mtcars)
+  replacement <- "i <- mtcars"
+  qr <- replace_code(qq, replacement)
+  before <- get_code(qq)[-length(get_code(qq))]
+  after <- get_code(qr)[-length(get_code(qr))]
+  testthat::expect_identical(before, after)
+})
```

Comment:
Replacement of the last element of the @code is exactly what `code_replace` does and other tests checking derivative information (length, last different, non-last identical)

```suggestion
testthat::test_that("code_replace replaces last element of the code", {
  qq <- within(qq, m <- mtcars)
  replacement <- "i <- mtcars"
  qr <- replace_code(qq, replacement)
  previous <- get_code(qq)
  current <- get_code(qr)
  testthat::expect_identical(current, c(head(previous, -1), replacement)
})
```